### PR TITLE
[MODORDERS-892] Using the new transaction batch endpoint

### PR DESCRIPTION
**Pull request by damien-git**
## Purpose
[MODORDERS-892](https://issues.folio.org/browse/MODORDERS-892) - Accumulate all transactions in holder to make only single call to mod-finance

## Approach
Used the new batch endpoint whenever possible and simplified. In `EncumbranceService.createOrUpdateEncumbrances()`, it was possible to gather several create/update/delete operations together with a single batch call.

#### TODOS and Open Questions
TODO: 
- use patch where it makes sense when it is implemented (see [MODORDERS-1008](https://issues.folio.org/browse/MODORDERS-1008)).
- add a sanity check in mod-finance-storage to make sure the transaction metadata property is present when updating transactions (will be done as part of [MODFISTO-432](https://issues.folio.org/browse/MODFISTO-432)).

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/models/EncumbrancesProcessingHolder.java [hunk 1]
```diff
@@ -113,10 +104,7 @@ public List<Transaction> getEncumbrancesToUnreleaseAfter() {
 
   public int getAllEncumbrancesQuantity() {
     return encumbrancesForCreate.size() + encumbrancesForUpdate.size()
-        + encumbrancesForRelease.size() + encumbrancesForUnrelease.size();
-  }
-
-  public List<Transaction> getEncumbrancesFromStorage() {
-    return ImmutableList.copyOf(encumbrancesFromStorage);
+        + encumbrancesForRelease.size() + encumbrancesForUnrelease.size()
+        + encumbrancesForDelete.size();
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/models/EncumbrancesProcessingHolder.java:108]: can you please explain more about this change, it was some bug for it previously?

File: src/main/java/org/folio/models/EncumbrancesProcessingHolder.java [hunk 2]
```diff
@@ -63,11 +59,6 @@ public EncumbrancesProcessingHolder withEncumbrancesForRelease(List<Transaction>
     return this;
   }
 
-  public EncumbrancesProcessingHolder withEncumbrancesFromStorage(List<Transaction> encumbrancesFromStorage) {
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/models/EncumbrancesProcessingHolder.java:66]: we deleted because it is not used any more?

Comment (damien-git) [reply src/main/java/org/folio/models/EncumbrancesProcessingHolder.java:66]: Right. And this is unrelated to the other changes.

**Review by SerhiiNosko — APPROVED** [review]
Code was simplified a lot, thanks @damien-git

**Review by imerabishvili — APPROVED** [review]

**Pull request merged**